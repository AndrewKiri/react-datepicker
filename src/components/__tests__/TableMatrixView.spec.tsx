import * as React from 'react';
import * as sinon from 'sinon';
import { shallow } from 'enzyme';
import TableMatrixView from '../TableMatrixView';

describe('<TableMatrixView/>', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('props matrix correctly', () => {
    const arr = [['1', '2', '3'], ['4', '5', '6'], ['7', '8', '9']];
    const component = shallow(
      <TableMatrixView
        matrix={arr}
        cell={(value: string, key: number) => <td key={key}>{value}</td>}
      />
    );
    expect(component).toMatchSnapshot();
    expect(component.find('td')).toHaveLength(9);
  });

  it('props headers correctly', () => {
    const header = ['1', '2', '3'];
    const arr = [['1', '2', '3'], ['4', '5', '6'], ['7', '8', '9']];
    const component = shallow(
      <TableMatrixView
        headers={header}
        matrix={arr}
        cell={(value: string, key: number) => <td key={key}>{value}</td>}
      />
    );

    expect(component.find('thead tr').children()).toHaveLength(3);
  });
});
