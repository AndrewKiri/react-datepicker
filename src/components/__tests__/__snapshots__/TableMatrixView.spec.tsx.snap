// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableMatrixView/> props matrix correctly 1`] = `
<table
  className="calendar__body--table"
>
  <tbody>
    <tr
      key="0"
    >
      <td
        key="0"
      >
        1
      </td>
      <td
        key="1"
      >
        2
      </td>
      <td
        key="2"
      >
        3
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        key="1"
      >
        4
      </td>
      <td
        key="2"
      >
        5
      </td>
      <td
        key="3"
      >
        6
      </td>
    </tr>
    <tr
      key="2"
    >
      <td
        key="2"
      >
        7
      </td>
      <td
        key="3"
      >
        8
      </td>
      <td
        key="4"
      >
        9
      </td>
    </tr>
  </tbody>
</table>
`;
